app/src/main/java/com/example/hw14bottomnavigation/RecyclerView/PetListAdapter.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.hw14bottomnavigation.RecyclerView
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.example.hw14bottomnavigation.RoomDatabase.RoomPet
+import com.example.hw14bottomnavigation.databinding.ItemPetBinding
+
+class PetViewHolder(
+    private val binding: ItemPetBinding
+) : RecyclerView.ViewHolder(binding.root) {
+    fun bind(pet: RoomPet) {
+        with(binding) {
+            petId.text = pet.id.toString()
+            petType.text = pet.petType
+            petBreed.text = pet.petBreed
+            petName.text = pet.petName
+        }
+    }
+}
+
+class PetListAdapter: ListAdapter<RoomPet, PetViewHolder>(DIFF_UTIL) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PetViewHolder {
+        return PetViewHolder(
+            ItemPetBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent,
+                false
+            )
+        )
+    }
+
+    override fun onBindViewHolder(holder: PetViewHolder, position: Int) {
+        holder.bind(getItem(position))
+    }
+
+    companion object {
+        private val DIFF_UTIL = object : DiffUtil.ItemCallback<RoomPet>() {
+            override fun areItemsTheSame(oldItem: RoomPet, newItem: RoomPet): Boolean {
+                return oldItem == newItem
```

Comment:
здесь должна быть легкая проверка, что элементы совпадают, какой-то признак уникальности элемента, обычно это id
```suggestion
                return oldItem .id== newItem.id
```